pkg/git/libgit2/managed/ssh.go

```diff
@@ -158,7 +160,14 @@ func (t *sshSmartSubtransport) Action(transportOptionsURL string, action git2go.
 	if u.Port() != "" {
 		port = u.Port()
 	}
-	t.addr = net.JoinHostPort(u.Hostname(), port)
+	addr := net.JoinHostPort(u.Hostname(), port)
+
+	if t.ctx == nil {
+		t.ctx = opts.Context
```

Comment:
The CI seems to be failing because the test doesn't sets the context in https://github.com/fluxcd/source-controller/blob/1faa547dad518ef88798ae6791c3ebb467c87aad/pkg/git/libgit2/managed/ssh_test.go#L103 , and this opts.Context remains nil.

Reply:
Fixed this, now both `ctx` and `logger` have safe defaults.